tokenizers/src/models/bpe/mod.rs

```diff
@@ -9,65 +9,40 @@ mod word;
 type Pair = (u32, u32);
 
 /// Errors that can be encountered while using or constructing a `BPE` model.
-#[derive(Debug)]
+#[derive(thiserror::Error, Debug)]
 pub enum Error {
     /// An error encountered while reading files mainly.
-    Io(std::io::Error),
+    #[error("IoError: {0}")]
+    Io(
+        #[from]
+        #[source]
```

Comment:
> The #[from] attribute always implies that the same field is #[source], so you don't ever need to specify both attributes.

https://github.com/dtolnay/thiserror

Do we really need `#[source]` here ?

Reply:
I think the macro derives that automatically though, doesn't it ?

Reply:
according to this rule, it will not (unless I'm misreading it)
> The Error trait's source() method is implemented to return whichever field has a #[source] attribute or is named source, if any. This is for identifying the underlying lower level error that caused your error.

Reply:
https://github.com/huggingface/tokenizers/pull/951/commits/3856dbce6b74c4be9472b90cdfc5b313cbf4a951